packages/kbn-shared-ux-components/src/empty_state/no_data_views/no_data_views.tsx

```diff
@@ -7,15 +7,13 @@
  */
 
 import React, { useCallback, useEffect, useRef } from 'react';
-
-import { DataView } from '../../../../../data_views/public';
-import { useEditors, usePermissions } from '../../../services';
-import type { SharedUXEditorsService } from '../../../services/editors';
+import { useEditors, usePermissions } from '@kbn/shared-ux-services';
+import type { SharedUXEditorsService } from '@kbn/shared-ux-services';
 
 import { NoDataViews as NoDataViewsComponent } from './no_data_views.component';
 
 export interface Props {
-  onDataViewCreated: (dataView: DataView) => void;
+  onDataViewCreated: (dataView: unknown) => void;
```

Comment:
Since we're in the process of migrating to packages, having a `types` package is a pretty standard deal. We should really work on getting one for common types, at least the ones from  `data` and `data_view` plugins. Is this on your radar as well @spalger ?

Reply:
I'm not sure if that would accomplish more than colocating the types today.  Technically, Bazel already builds a separate types "package"... and I'd be worried about keeping types too far from the relevant code.

At the same time, I'd be interested in creating a `@kbn/shared-ux-types` package for common and public types.  I'd just want to keep it curated to truly universal/common types, rather than default to putting all types there.

It's a good point.  Let's think on it a bit, see what we'd propose to put in a `types` package to see if there'd be value in it.  @spalger @stacey-gammon I'd be curious on your input here, as well.

Reply:
My opinion is to stick to our '"organize by domain" philosophy and co-locate types with the thing they are describing (meaning `DataView` types should be next to `DataView` implementation stuff).

Am I understanding the issue here is that packages can't depend on plugins and thats why this had to change from a type to `unknown`?

I think we will eventually be able to solve this when each plugin is a package, and there is no separate `packages` directory at all (https://github.com/elastic/kibana/issues/112886)

cc @mistic

Reply:
Yeah, I feel like shared-ux is just a little ahead of the curve here. The DataView plugin will eventually be a package and `@kbn/shared-ux` will be able to import the `DataView` type from it, a `@kbn/data-views-types` package, or something similar but co-located with the data views implementation stuff.

Reply:
>  I'd just want to keep it curated to truly universal/common types, rather than default to putting all types there.

> My opinion is to stick to our '"organize by domain" philosophy and co-locate types with the thing they are describing

Totally agree +++